src/servant/tt_004/sword.c

```diff
@@ -0,0 +1,545 @@
+            self->step++;
+            break;
+        }
+    }
+
+    D_us_80178B80 = 0;
+    self->ext.swordFamiliar.unk7c = self->entityId;
+}
+
+// chooses the largest magnitude of arg0, arg1, arg2
+//
+// see: Min3
+static s32 Max3(s32 arg0, s32 arg1, s32 arg2) {
+    arg0 = abs(arg0);
+    arg1 = abs(arg1);
+    arg2 = abs(arg2);
+
+    if (arg0 >= arg1 && arg0 >= arg2) {
+        return arg0;
+    }
+
+    if (arg1 >= arg0 && arg1 >= arg2) {
+        return arg1;
+    }
+
+    if (arg2 >= arg0 && arg2 >= arg1) {
+        return arg2;
+    }
+}
+
+// chooses the smallest magnitude of arg0, arg1, arg2
+//
+// this function is unused.
+//
+// see: Max3
+static s32 Min3(s32 arg0, s32 arg1, s32 arg2) {
+    arg0 = abs(arg0);
+    arg1 = abs(arg1);
+    arg2 = abs(arg2);
+
+    if (arg0 <= arg1 && arg0 <= arg2) {
+        return arg0;
+    }
+
+    if (arg1 <= arg0 && arg1 <= arg2) {
+        return arg1;
+    }
+
+    if (arg2 <= arg0 && arg2 <= arg1) {
+        return arg2;
+    }
+}
+
+typedef struct {
+    s16 unk0;
+    s16 unk2;
+    s16 unk4;
+    s16 unk6;
+    s16 unk8;
+    s16 unkA;
+    s16 unkC;
+    s16 unkE;
+    s16 unk10;
+} UnkC8C;
+
+void func_us_80172C8C(UnkC8C* arg0, UnkC8C* arg1) {
+    D_us_801782B0 = rsin(arg0->unk0);
+    D_us_801782B4 = rsin(arg0->unk2);
+    D_us_801782B8 = rsin(arg0->unk4);
+    D_us_801782BC = rcos(arg0->unk0);
+    D_us_801782C0 = rcos(arg0->unk2);
+    D_us_801782C4 = rcos(arg0->unk4);
+
+    arg1->unk0 = (D_us_801782C4 * D_us_801782C0) >> 0xC;
+    arg1->unk2 =
+        ((-D_us_801782B8 * D_us_801782BC) >> 0xC) +
+        ((D_us_801782B0 * ((D_us_801782C4 * D_us_801782B4) >> 0xC)) >> 0xC);
+    arg1->unk4 =
+        ((D_us_801782B8 * D_us_801782B0) >> 0xC) +
+        ((D_us_801782BC * ((D_us_801782C4 * D_us_801782B4) >> 0xC)) >> 0xC);
+    arg1->unk6 = (D_us_801782B8 * D_us_801782C0) >> 0xC;
+    arg1->unk8 =
+        ((D_us_801782C4 * D_us_801782BC) >> 0xC) +
+        ((D_us_801782B0 * ((D_us_801782B8 * D_us_801782B4) >> 0xC)) >> 0xC);
+    arg1->unkA =
+        ((-D_us_801782C4 * D_us_801782B0) >> 0xC) +
+        ((D_us_801782BC * ((D_us_801782B8 * D_us_801782B4) >> 0xC)) >> 0xC);
+    arg1->unkC = -D_us_801782B4 >> 0xC;
+    arg1->unkE = (D_us_801782C0 * D_us_801782B0) >> 0xC;
+    arg1->unk10 = (D_us_801782C0 * D_us_801782BC) >> 0xC;
+}
+
+INCLUDE_ASM("servant/tt_004/nonmatchings/sword", func_us_80172E84);
+
+extern s32 D_us_80178344[];
+extern s32 D_us_80178B88;
+
+Entity* func_us_80173AA0(Entity* self) {
+    Entity* entity;
+    s32 i;
+    s32 entityIndex;
+    s32 hitFlag;
+
+    hitFlag = 0;
+    entity = &g_Entities[0x40];
+
+    for (i = 0; i < 128; i++, entity++) {
+        D_us_80178344[i] = 0;
+        if (entity->entityId && entity->hitboxState != 0 &&
+            !(entity->flags & FLAG_UNK_00200000) &&
+            *((s16*)&entity->posX.i.hi) >= -16 && entity->posX.i.hi <= 272 &&
+            *((s16*)&entity->posY.i.hi) <= 240 && entity->posY.i.hi >= 0 &&
```

Comment:
`Entity.posX.i.hi` is an `s16` but without the cast the value was being loaded with `lhu`. This doesn't seem like it should be necessary, but I couldn't find a pattern that would match, otherwise.